galley/pkg/server/components/processing2.go

```diff
@@ -0,0 +1,348 @@
+
+	serveWG       sync.WaitGroup
+	grpcServer    *grpc.Server
+	runtime       *processing.Runtime
+	mcpSource     *source.Server
+	reporter      monitoring.Reporter
+	callOut       *callout
+	listenerMutex sync.Mutex
+	listener      net.Listener
+	stopCh        chan struct{}
+}
+
+var _ process.Component = &Processing2{}
+
+// NewProcessing2 returns a new processing component.
+func NewProcessing2(a *settings.Args) *Processing2 {
+	d := snapshot.New(groups.IndexFunction)
+	return &Processing2{
+		args:         a,
+		distributor:  d,
+		configzTopic: configz.CreateTopic(d),
+	}
+}
+
+// Start implements process.Component
+func (p *Processing2) Start() (err error) {
+	var mesh event.Source
+	var src event.Source
+
+	if mesh, err = meshcfgNewFS(p.args.MeshConfigFile); err != nil {
+		return
+	}
+
+	m := metadata.MustGet()
+
+	kubeResources := p.disableExcludedKubeResources(m)
+
+	if src, err = p.createSource(kubeResources); err != nil {
+		return
+	}
+
+	if p.runtime, err = processorInitialize(m, p.args.DomainSuffix, event.CombineSources(mesh, src), p.distributor); err != nil {
+		return
+	}
+
+	grpcOptions := p.getServerGrpcOptions()
+
+	p.stopCh = make(chan struct{})
+	var checker source.AuthChecker = server.NewAllowAllChecker()
+	if !p.args.Insecure {
+		if checker, err = watchAccessList(p.stopCh, p.args.AccessListFile); err != nil {
+			return
+		}
+
+		var watcher creds.CertificateWatcher
+		if watcher, err = creds.PollFiles(p.stopCh, p.args.CredentialOptions); err != nil {
+			return
+		}
+		credentials := creds.CreateForServer(watcher)
+
+		grpcOptions = append(grpcOptions, grpc.Creds(credentials))
+	}
+	grpc.EnableTracing = p.args.EnableGRPCTracing
+	p.grpcServer = grpc.NewServer(grpcOptions...)
+
+	p.reporter = mcpMetricReporter("galley/mcp/source")
+
+	options := &source.Options{
+		Watcher:            p.distributor,
+		Reporter:           p.reporter,
+		CollectionsOptions: source.CollectionOptionsFromSlice(m.SnapshottedCollectionNames()),
+		ConnRateLimiter:    mcprate.NewRateLimiter(time.Second, 100), // TODO(Nino-K): https://github.com/istio/istio/issues/12074
+	}
+
+	if p.args.SinkAddress != "" {
+		p.callOut, err = newCallout(p.args.SinkAddress, p.args.SinkAuthMode, p.args.SinkMeta, options)
+		if err != nil {
+			p.callOut = nil
+			scope.Errorf("Callout could not be initialized: %v", err)
+			return
+		}
+	}
+
+	serverOptions := &source.ServerOptions{
+		AuthChecker: checker,
+		RateLimiter: rate.NewLimiter(rate.Every(time.Second), 100), // TODO(Nino-K): https://github.com/istio/istio/issues/12074
+	}
+
+	p.mcpSource = source.NewServer(options, serverOptions)
+
+	// get the network stuff setup
+	network := "tcp"
+	var address string
+	idx := strings.Index(p.args.APIAddress, "://")
+	if idx < 0 {
+		address = p.args.APIAddress
+	} else {
+		network = p.args.APIAddress[:idx]
+		address = p.args.APIAddress[idx+3:]
+	}
+
+	if p.listener, err = netListen(network, address); err != nil {
+		err = fmt.Errorf("unable to listen: %v", err)
```

Comment:
Above failure logs, this one doesn't. On purpose?

Reply:
Not particularly. The error already gets logged after it is bubbled up. The log here is redundant. (hazards of copy paste). Removing the error logs.